### ISSUE TITLE
extend Enum api

### ISSUE THREAD

**Issue by Borda** [PyTorchLightning/pytorch-lightning#5478]
## 🚀 Feature

it may be nice to have some general calls over our enums such as Distributed mentioned by @carmocca in https://github.com/PyTorchLightning/pytorch-lightning/pull/5300#discussion_r555269347

Any logic to compare different `DistributedType`s should be encapsulated by the enum itself. 

- [x] https://github.com/PyTorchLightning/pytorch-lightning/pull/5743#discussion_r574668811
- [x] https://github.com/PyTorchLightning/pytorch-lightning/pull/5743#discussion_r574669909
- [x] https://github.com/PyTorchLightning/pytorch-lightning/pull/5970#discussion_r578829098

### Motivation

easier handling some cumulative behaviours, such `is_ddp` which cover `DDP` and `DDP_SPAWN`

### Pitch

```python
class DistributedType(LightningEnum):
    DP = 'dp'
    ...

    def is_distributed(self):
        return self in (DistributedType.DDP, DistributedType.DDP_SPAWN, DistributedType.DDP2)
```

**Comment by awaelchli** [comment]
I would like to do a comment/counter-pitch here.
If we had a strict, finite set of training types the enum with the is_distributed etc. functionality would be a great solution. 
However, users can pass in their own plugins and they will not fall into the fixed categories
`(DistributedType.DDP, DistributedType.DDP_SPAWN, DistributedType.DDP2)`

I suggest that we keep these enum types strictly internal and only use them for argument parsing part where the options are strictly limited.

**Comment by carmocca** [comment]
Maybe I am missing something but isn't that already a problem? see (random example):

https://github.com/PyTorchLightning/pytorch-lightning/blob/6b990f3fa5646cfbf2e2945daf568a27921b8362/pytorch_lightning/trainer/connectors/accelerator_connector.py#L518

This issue is just about moving those checks from one place to the other (connector to Enum), not if those checks should exist at all

**Comment by awaelchli** [comment]
Yes, should be fine to move them.
Just wanted to bring awareness of the current limitation

**Comment by carmocca** [comment]
This might not be relevant after #6090, wdyt @kaushikb11 ?

**Comment by kaushikb11** [comment]
Interesting, this is something that could be looked into. Like the idea of having it separate from the Accelerator Connector.

**Comment by carmocca** [comment]
Closing as we are trying to reduce our dependence on Enums, since they limit customization: #10422
